packages/core-data/src/actions.js

```diff
@@ -115,14 +115,97 @@ export function receiveEmbedPreview( url, preview ) {
 	};
 }
 
+/**
+ * Returns an action object that triggers an
+ * edit to an entity record.
+ *
+ * @param {string} kind           Kind of the edited entity record.
+ * @param {string} name           Name of the edited entity record.
+ * @param {number} recordId       Record ID of the edited entity record.
+ * @param {Object} edits          The edits.
+ *
+ * @return {Object} Action object.
+ */
+export function* editEntityRecord( kind, name, recordId, edits ) {
+	const { transientEdits = {} } = yield select( 'getEntity', kind, name );
+	const record = yield select( 'getEntityRecord', kind, name, recordId );
+	const editedRecord = yield select(
+		'getEditedEntityRecord',
+		kind,
+		name,
+		recordId
+	);
+
+	const edit = {
+		kind,
+		name,
+		recordId,
+		// Clear edits when they are equal to their persisted counterparts
+		// so that the property is not considered dirty.
+		edits: Object.keys( edits ).reduce( ( acc, key ) => {
+			acc[ key ] = isEqual( record[ key ], edits[ key ] ) ? undefined : edits[ key ];
+			return acc;
+		}, {} ),
+		transientEdits,
```

Comment:
This is already available in the reducer, why duplicate it in the action? (I guess to ease access to this value in a sub reducer), can't we leverage a higher-order reducer to add it to the action instead?

Reply:
>can't we leverage a higher-order reducer to add it to the action instead

Good idea, I'll do that.